scripts/language_model/train/transformer_lm.py

```diff
@@ -0,0 +1,233 @@
+    drop_e : float
+        Dropout rate to use on the embedding layer.
+    drop_l : float
+        Dropout rate to use on the latent layer.
+    num_experts : int
+        Number of experts in mixture of softmax.
+    upperbound_fixed_layer : int
+        Number of layers in BERT with the parameters fixed.
+
+    Inputs:
+        - **inputs**: input sequence tensor, shape (batch_size, seq_length)
+        - **begin_state**: initial recurrent state tensor with length equals to num_layers.
+            the initial state with shape `(1, batch_size, num_hidden)`
+        - **token_types**: input token type tensor, shape (batch_size, seq_length).
+            If the inputs contain two sequences, then the token type of the first
+            sequence differs from that of the second one.
+        - **valid_length**: optional tensor of input sequence valid lengths, shape (batch_size,)
+        - **masked_positions**: optional tensor of position of tokens for masked LM decoding,
+            shape (batch_size, num_masked_positions).
+
+    Outputs:
+        - **out**: output tensor with shape `(sequence_length, batch_size, input_size)`
+            when `layout` is "TNC".
+        - **out_states**: output recurrent state tensor with length equals to num_layers.
+            the state with shape `(1, batch_size, num_hidden)`
+        - **encoded_raw**: The list of outputs of the model's encoder with length equals to
+            num_layers. the shape of every encoder's output
+            `(sequence_length, batch_size, num_hidden)`
+        - **encoded_dropped**: The list of outputs with dropout of the model's encoder with
+            length equals to num_layers. The shape of every encoder's dropped output
+            `(sequence_length, batch_size, num_hidden)`
+    """
+
+    def __init__(self, embedding, mode, vocab_size, embed_size=300, hidden_size=1150, hidden_size_last=650,
+                 num_layers=3, tie_weights=True, dropout=0.4, weight_drop=0.5, drop_h=0.2,
+                 drop_i=0.55, drop_e=0.1, drop_l=0.29, num_experts=15,
+                 upperbound_fixed_layer=22, **kwargs):
+        super(BERTRNN, self).__init__(**kwargs)
+        self._mode = mode
+        self._vocab_size = vocab_size
+        self._hidden_size = hidden_size
+        self._hidden_size_last = hidden_size_last
+        self._num_layers = num_layers
+        self._dropout = dropout
+        self._drop_h = drop_h
+        self._drop_i = drop_i
+        self._drop_e = drop_e
+        self._drop_l = drop_l
+        self._num_experts = num_experts
+        self._weight_drop = weight_drop
+        self._tie_weights = tie_weights
+
+        #bert specific setting
+        self._embed_size = embedding.encoder._units
+
+        with self.name_scope():
+            self.embedding = embedding
+            for cell in self.embedding.encoder.transformer_cells[:upperbound_fixed_layer]:
+                cell.collect_params().setattr('grad_req', 'null')
```

Comment:
It seems that this could have been done outside the class with more flexibility than what the `upperbound_fixed_layer` could provide. Since there's no pretraining script in this PR, it might be better to remove the training related argument for now.